packages/react-components/src/components/table-output-component.tsx

```diff
@@ -653,4 +655,126 @@ export class TableOutputComponent extends AbstractOutputComponent<TableOutputPro
             });
         }
     }
+
+    async exportOutput(option: string): Promise<void> {
+        if (!this.gridApi || !this.columnApi) {
+            return;
+        }
+
+        const traceUUID = this.props.traceId;
+        const tspClient = this.props.tspClient;
+        const outputId = this.props.outputDescriptor.id;
+        let fetchIndex = 0;
+        let totalLinesToFetch = this.gridApi.getDisplayedRowCount();
+        const linesArray = new Array<Line>();
+        const bufferSize = 100000;
+        const columnsIds: Array<number> = [];
+
+        // Convert to csv
+        const csv_data = [];
+
+        // Fetch column headers
+        const colHeaderRow: string[] = [];
+
+        const columns = this.columnApi?.getAllColumns();
+        columns.forEach((column, index) => {
+            if (column.isVisible()) {
+                colHeaderRow.push(column.getColDef().headerName ?? 'no header');
+                columnsIds.push(index);
+            }
+        });
+        csv_data.push(colHeaderRow.join(','));
+
+        if (option === 'selection') {
+            if (this.enableIndexSelection && this.selectStartIndex !== -1 && this.selectEndIndex !==-1) {
+                fetchIndex = Math.min(this.selectStartIndex, this.selectEndIndex);
+                totalLinesToFetch = Math.abs(this.selectStartIndex - this.selectEndIndex) + 1;
+            } else if (!this.enableIndexSelection && this.startTimestamp && this.endTimestamp) {
+                const startIndex = await this.fetchTableIndex(this.startTimestamp < this.endTimestamp ? this.startTimestamp : this.endTimestamp);
+                const endIndex = await this.fetchTableIndex(this.startTimestamp > this.endTimestamp ? this.startTimestamp : this.endTimestamp);
+
+                if (startIndex && endIndex) {
+                    fetchIndex = startIndex;
+                    totalLinesToFetch = endIndex - startIndex;
+                }
+            }
+        }
+
+        let fetchLinesRemainder = totalLinesToFetch;
+
+        while (fetchLinesRemainder > 0) {
+            let curLinesToFetch = bufferSize;
+
+            if (fetchLinesRemainder > bufferSize) {
+                fetchLinesRemainder-=bufferSize;
+            } else {
+                curLinesToFetch = fetchLinesRemainder;
+                fetchLinesRemainder = 0;
+            }
+
+            const tspClientResponse = await tspClient.fetchTableLines(traceUUID, outputId, QueryHelper.tableQuery(this.columnIds, fetchIndex, curLinesToFetch,
```

Comment:
The fetchIndex needs to be incremented in the loop! It's always fetching the first 10000 lines at every iteration...

If you have fetchIndex properly used, you might not need fetchLinesRemainder.